packages/eslint-plugin/src/utils/helpers.ts

```diff
@@ -0,0 +1,170 @@
+import type { RuleContext } from '@typescript-eslint/utils/ts-eslint'
+import type { ImportResult } from '@pandacss/core'
+import type { TSESTree } from '@typescript-eslint/utils'
+import { AST_NODE_TYPES } from '@typescript-eslint/utils'
+import { syncAction } from './'
+
+const getAncestorOfType = <
+  T extends TSESTree.Node,
+  A extends AST_NODE_TYPES,
+  Node = Extract<TSESTree.Node, { type: A }>,
+>(
+  node: T,
+  type: A,
+): Node | undefined => {
+  let current: TSESTree.Node | undefined = node.parent
+  while (current) {
+    if (current.type === type) return current as Node
+    current = current.parent
+  }
+
+  return
+}
+
+const getSyncOpts = (context: RuleContext<any, any>) => {
+  return {
+    currentFile: context.getFilename(),
+    configPath: context.settings['pandacss/configPath'] as string | undefined,
+  }
+}
+
+const getImports = (context: RuleContext<any, any>) => {
+  const imports: ImportResult[] = []
+
+  context.getSourceCode().ast.body.forEach((node) => {
+    if (node.type !== 'ImportDeclaration') return
+
+    const mod = node.source.value
+    if (!mod) return
+
+    node.specifiers.forEach((specifier) => {
+      if (specifier.type !== 'ImportSpecifier') return
+
+      const name = specifier.imported.name
+      const alias = specifier.local.name
+
+      const result = { name, alias, mod }
+
+      const found = syncAction('matchImports', getSyncOpts(context), result)
+
+      if (!found) return
+
+      imports.push(result)
+    })
+  })
+
+  return imports
+}
+
+const isValidStyledProp = <T extends TSESTree.Node | string>(node: T, context: RuleContext<any, any>) => {
+  if (typeof node === 'string') return
+  return node.type === 'JSXIdentifier' && isValidProperty(node.name, context)
+}
+
+const isPandaIsh = (name: string, context: RuleContext<any, any>) => {
+  const imports = getImports(context)
+  return syncAction('matchFile', getSyncOpts(context), name, imports)
+}
+
+// TODO - We might want to parse the AST from the root to find the declaration
+const findDeclaration = (name: string, context: RuleContext<any, any>) => {
```

Comment:
I'm not sure if those comments are still relevant but I implemented something related in the extractor 
https://github.com/chakra-ui/panda/blob/main/packages/extractor/src/find-identifier-value-declaration.ts#L78